site/public/js/images.js

```diff
@@ -1,20 +1,19 @@
 $(document).ready(function(){
 
 $('.no-image-available').each(function(i, obj) {
+	var student_name = $(obj).data('student-name');
+	// Identicon requires a harsh of at least 15 chars
+	// So to prevent short names from breaking it, the name is salted.
+	// This would ensure resulting harshes are longer than 15 chars.
+	student_name = student_name+'0123456789_abcdefghijklmnopqrstuvwxyz';
     // create a base64 encoded PNG
-    var data = new Identicon(randomString(32), 420).toString();
-    // write to a data URI
+    var data = new Identicon(uniqueHarsh(student_name), 420).toString();
```

Comment:
remove the one-line `uniqueHarsh` function and just do the `btoa(student_name)` here.

Reply:
Alright, thought of this but... doing that right away. Thanks.